core/src/worker/activities/local_activities.rs

```diff
@@ -70,7 +74,18 @@ impl LocalActivityExecutionResult {
         Self::Cancelled(Cancellation::from_details(None))
     }
     pub(crate) fn timeout(tt: TimeoutType) -> Self {
-        Self::TimedOut(ActFail::timeout(tt))
+        Self::TimedOut(ActFail {
+            failure: Some(APIFailure {
+                message: "Activity timed out".to_string(),
+                failure_info: Some(failure::FailureInfo::TimeoutFailureInfo(
```

Comment:
AFAIU this should be an ActivityFailure with TimeoutFailure set as the cause.

Reply:
Thanks - this isn't quite the right spot but I added it to the right place